### PR TITLE
f

### PR DESCRIPTION
**Pull request by XOctOpus1**

### CODE REVIEW

File: app/customer.py [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+
+    def get_shop(self, shops: List[Shop], fuel_price: int | float) -> None:
+        self_x, self_y = self.location
+        best_shop = {"total_cost": None, "shop": None}
+
+        for shop in shops:
+            total_cost = 0
+            total_shop = best_shop["total_cost"]
+            shop_x, shop_y = shop.location
+            distance = math.sqrt(
```
Comment (vasylhnatiuk) [inline app/customer.py:46]: distance = math.sqrt(
                (self_x - shop_x) ** 2 + (self_y - shop_y) ** 2
            ) better write in a separate function.

File: app/customer.py [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+            total_cost += self.__calculate_money_for_products(shop.products)
+
+            print(f"{self.name}'s trip to the {shop.name} costs {total_cost}")
+
+            if total_cost <= self.money and (
+                    not total_shop or total_shop > total_cost
+            ):
+                best_shop = {"total_cost": total_cost, "shop": shop}
+        if best_shop["shop"]:
+            total_cost, shop = best_shop.values()
```
Comment (vasylhnatiuk) [inline app/customer.py:59]: total_cost, shop = best_shop.values()
            print(f"{self.name} rides to {shop.name}\n")
            shop.bill(self.name, self.product_cart)
            print(f"{self.name} rides home")
            print(f"{self.name} now has {self.money - total_cost} dollars\n") better write in separate function

**Review by vasylhnatiuk — CHANGES_REQUESTED** [review]
`When the customer arrives at the shop his location should equal to shop location. After customer buys products, shop prints purchase receipt using current time. After the shop he arrives home`
you didn't do it

File: app/customer.py (hunk 2 above)
Comment (XOctOpus1) [reply app/customer.py:59]: добрий вечір, не сильно розумію що значить писати в роздільних функціях

**Review by ArsenPidhoretskyi — APPROVED** [review]
Well done!

**Pull request closed without merging**